File: csharp/ToolGood.Algorithm/Internals/Functions/MathSum/Function_FISHERINV.cs
using System;
using System.Text;

namespace ToolGood.Algorithm.Internals.Functions.MathSum
{
	internal class Function_FISHERINV : Function_1
    {
        public Function_FISHERINV(FunctionBase func1) : base(func1)
        {
        }

        public override string Name => "FisherInv";

        public override Operand Evaluate(AlgorithmEngine work, Func<AlgorithmEngine, string, Operand> tempParameter)
        {
            var args1 = func1.Evaluate(work, tempParameter);
            args1 = ConvertToNumber(args1, "FisherInv", 1);
            if (args1.IsError) { return args1; }
            var x = args1.DoubleValue;
            var n = (Math.Exp((2 * x)) - 1) / (Math.Exp((2 * x)) + 1);
            return Operand.Create(n);
        }

    }

}

File: java/src/main/java/toolgood/algorithm/internals/functions/mathsum/Function_FISHERINV.java
package toolgood.algorithm.internals.functions.mathsum;

import toolgood.algorithm.internals.Operand;
import toolgood.algorithm.internals.FunctionBase;
import toolgood.algorithm.internals.Function_1;
import toolgood.algorithm.internals.AlgorithmEngine;

public class Function_FISHERINV extends Function_1 {
    public Function_FISHERINV(FunctionBase func1) {
        super(func1);
    }

    @Override
    public Operand Evaluate(AlgorithmEngine work, java.util.function.Function<AlgorithmEngine, String, Operand> tempParameter) {
        Operand args1 = func1.Evaluate(work, tempParameter);
        if (args1.IsNotNumber()) {
            args1 = args1.ToNumber("Function '{0}' parameter is error!", "FisherInv");
            if (args1.IsError()) {
                return args1;
            }
        }
        double x = args1.DoubleValue();
        double n = (Math.exp(2 * x) - 1) / (Math.exp(2 * x) + 1);
        return Operand.Create(n);
    }

    @Override
    public void toString(java.lang.StringBuilder stringBuilder, boolean addBrackets) {
        AddFunction(stringBuilder, "FisherInv");
    }
}
